src/tests/test_magic_cmd.py

```diff
@@ -605,3 +663,34 @@ def test_delete_invalid_snippet(arg, ip_snippets):
 
     assert excinfo.value.error_type == "UsageError"
     assert str(excinfo.value) == "No such saved snippet found : non_existent_snippet"
+
+
+def test_table_explore_with_snippets(ip, tmp_empty):
+    ip.run_cell(
+        """
+        %%sql duckdb://
```

Comment:
if you pass duckdb:// you'll overwrite the db and you'll be testing with duckdb instead of the database from the ip_with_dynamic_db, please remove